### PR TITLE
Feat/shr minor fixes

### PR DESCRIPTION
**Pull request by MohamedAliBouhaouala**
## Type of PR (check all applicable)

- [ ] 💼 New Package
- [x] ♻️ Refactor
- [ ] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🧪 Tests
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This PR addresses the SHR passthrough issue.

## Tickets & Related Documents

Resolves #

## Added code snippets?
- [ ] 👍 Yes
- [x] 🙅 No, because they aren't needed

## Added tests?

- [ ] 👍 Yes
- [x] 🙅 No, because they aren't needed
- [ ] 🙋 No, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📓 general documentation
- [x] 🙅 no documentation needed

## Are there any post-deployment tasks we need to perform?

-
-
-

### CODE REVIEW

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: projects/shr-mediator/src/botswana/controllers/kafka-consumer.controller.ts [hunk 1]
```diff
@@ -38,11 +38,13 @@ export class KafkaConsumerController {
...
   async handleSendAdtToIpms(
-    @Payload() val: string,
+    @Payload() val: IBundle,
     @Ctx() context: KafkaContext,
   ) {
     try {
-      const origBundle = JSON.parse(val).bundle;
+
+      //@ts-ignore
+      const origBundle: IBundle = val.bundle;
```
Comment (marrouchi) [inline projects/shr-mediator/src/botswana/controllers/kafka-consumer.controller.ts:47]: We need to avoid @ts-ignore, I think we can fix `@Payload() val: IBundle,` so that it returns the correct type

File: projects/shr-mediator/src/botswana/controllers/kafka-consumer.controller.ts [hunk 2]
```diff
@@ -114,7 +116,8 @@ export class KafkaConsumerController {
     @Ctx() context: KafkaContext,
   ) {
     try {
-      const origBundle = JSON.parse(val).bundle;
+      //@ts-ignore
```
Comment (marrouchi) [inline projects/shr-mediator/src/botswana/controllers/kafka-consumer.controller.ts:119]: Same here

File: projects/shr-mediator/src/botswana/controllers/kafka-consumer.controller.ts [hunk 3]
```diff
@@ -128,7 +131,8 @@ export class KafkaConsumerController {
     @Ctx() context: KafkaContext,
   ) {
     try {
-      const patient: IPatient = JSON.parse(val);
+      //@ts-ignore
```
Comment (marrouchi) [inline projects/shr-mediator/src/botswana/controllers/kafka-consumer.controller.ts:134]: Same here

File: projects/shr-mediator/src/botswana/services/hl7.service.ts [hunk 4]
```diff
@@ -175,16 +175,16 @@ export class Hl7Service {
...
             'content-type': 'text/plain',
           },
-          auth: {
-            username: config.get('mediator:client:username'),
-            password: config.get('mediator:client:password'),
-          },
+          // auth: {
+          //   username: config.get('mediator:client:username'),
+          //   password: config.get('mediator:client:password'),
+          // },
```
Comment (marrouchi) [inline projects/shr-mediator/src/botswana/services/hl7.service.ts:187]:
```suggestion

```

File: projects/shr-mediator/src/botswana/services/ipms.service.ts [hunk 5]
```diff
@@ -135,15 +136,38 @@ export class IpmsService {
...
+                  'fullUrl': fetchedBundle.entry[0].fullUrl,
+                  'resource' :<ILocation> fetchedBundle.entry[0].resource
+                }
+
+                this.logger.log(`Retrieved Location resource : ${JSON.stringify(locationResource)}`);
+                outBundle.entry.push(locationResource);
+          }
+        }
+       
+        // const translatedBundle = await this.labService.translateToTransactionBundle(outBundle);
```
Comment (marrouchi) [inline projects/shr-mediator/src/botswana/services/ipms.service.ts:165]: Do we still need this ?

File: projects/shr-mediator/src/botswana/services/ipms.service.ts [hunk 6]
```diff
@@ -391,14 +415,39 @@ export class IpmsService {
             : undefined;
 
         if (labOrderId && labOrderId.value) {
-          const options: AxiosRequestConfig = {
+        
+
+          const formerOptions: AxiosRequestConfig = {
```
Comment (marrouchi) [inline projects/shr-mediator/src/botswana/services/ipms.service.ts:420]: Are these still former ?

File: projects/shr-mediator/src/botswana/services/mfl.service.ts [hunk 7]
```diff
@@ -160,26 +157,26 @@ export class MflService {
...
-                url: orderingLocationRef.reference,
-              },
-            });
+            // bundle.entry.push({
+            //   resource: orderingLocation,
+            //   request: {
+            //     method: R4.Bundle_RequestMethodKind._put,
+            //     url: orderingLocationRef.reference,
+            //   },
+            // });
```
Comment (marrouchi) [inline projects/shr-mediator/src/botswana/services/mfl.service.ts:179]: Please add @todos here

File: projects/shr-mediator/src/botswana/services/mllp.service.ts [hunk 8]
```diff
@@ -16,7 +15,7 @@ export class MllpService implements OnModuleInit {
     this.mllpServer = new MllpServer(
       '0.0.0.0',
       config.get('app:mllpPort'),
-      logger,
+      // logger,
```
Comment (marrouchi) [inline projects/shr-mediator/src/botswana/services/mllp.service.ts:18]:
```suggestion
      // logger,
```
Can / Should we revert this change?

File: projects/shr-mediator/src/botswana/services/mllp.service.ts [hunk 9]
```diff
@@ -36,9 +35,7 @@ export class MllpService implements OnModuleInit {
       this.logger.debug('Received message:', data.toString());
       const checkChar: string = data[data.length - 1];
       if (checkChar == '\r') {
-        const response: IBundle = await this.hl7Handler.handleMessage(data);
-
-        this.logger.log('HL7 Response:\n' + JSON.stringify(response));
+        const response: any = await this.hl7Handler.handleMessage(data);
```
Comment (marrouchi) [inline projects/shr-mediator/src/botswana/services/mllp.service.ts:38]: Let's avoid "any"

File: projects/shr-mediator/src/botswana/services/mpi.service.ts [hunk 10]
```diff
@@ -31,12 +31,14 @@ export class MpiService {
...
         },
       };
+      const thePatient = patResult.resource as R4.IPatient
 
       const crResult = await this.fhirService.postWithRetry(
         patResult.resource as R4.IPatient,
         options,
         config.get('bwConfig:retryCount'),
         config.get('bwConfig:retryDelay'),
+        patResult.resource.resourceType
```
Comment (marrouchi) [inline projects/shr-mediator/src/botswana/services/mpi.service.ts:41]: Let's rather update the "postWithRetry()" function so that it uses the resourceType

File: projects/shr-mediator/src/common/services/fhir.service.ts [hunk 11]
```diff
@@ -101,17 +118,29 @@ export class FhirService {
...
     timeout = 30000,
+    path?: string
   ) {
     for (let attempt = 1; attempt <= retryLimit; attempt++) {
       try {
-        const { data } = await this.httpService.axiosRef.post<R4.IBundle>(
-          config.get('fhirServer:baseURL'),
+        if(!path){
+          path=''
+        }
```
Comment (marrouchi) [inline projects/shr-mediator/src/common/services/fhir.service.ts:127]: Not sure this is correct

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: projects/shr-mediator/src/common/services/fhir.service.ts (hunk 11 above)
Comment (MohamedAliBouhaouala) [reply projects/shr-mediator/src/common/services/fhir.service.ts:127]: To avoid 'undefined' exceptions

File: projects/shr-mediator/src/common/services/fhir.service.ts [hunk 12]
```diff
@@ -101,17 +118,29 @@ export class FhirService {
     options?: AxiosRequestConfig<any>,
     retryLimit = 2,
     timeout = 30000,
+    path?: string
```
Comment (marrouchi) [inline projects/shr-mediator/src/common/services/fhir.service.ts:121]:
```suggestion
    path?: string = ''
```
Let's rather do this.

File: projects/shr-mediator/src/common/services/fhir.service.ts (hunk 11 above)
Comment (marrouchi) [reply projects/shr-mediator/src/common/services/fhir.service.ts:127]: Let's rather do this. 
`path?: string = ''`

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: projects/shr-mediator/src/botswana/controllers/kafka-consumer.controller.ts (hunk 1 above)
Comment (MohamedAliBouhaouala) [reply projects/shr-mediator/src/botswana/controllers/kafka-consumer.controller.ts:47]: export interface BundlePayload {
  bundle: IBundle;
}
@Payload() val: BundlePayload

File: projects/shr-mediator/src/botswana/services/mllp.service.ts (hunk 8 above)
Comment (MohamedAliBouhaouala) [reply projects/shr-mediator/src/botswana/services/mllp.service.ts:18]: "TypeError: this.logger.info is not a function.". This is the error encountered when the logger is added to the constructor.

File: projects/shr-mediator/src/botswana/services/ipms.service.ts (hunk 6 above)
Comment (MohamedAliBouhaouala) [reply projects/shr-mediator/src/botswana/services/ipms.service.ts:420]: formerOptions is for fetching the previous Service Request resource. options (current options) is for fetching the task resource using include Search based on the retrieved SR resource.
